src/datascience-ui/ipywidgets/kernel/index.ts

```diff
@@ -159,19 +156,25 @@ function renderIPyWidget(
 let widgetManagerPromise: Promise<WidgetManager> | undefined;
 async function getWidgetManager(): Promise<WidgetManager> {
     if (!widgetManagerPromise) {
-        // eslint-disable-next-line , @typescript-eslint/no-explicit-any
-        widgetManagerPromise = new Promise((resolve) => WidgetManager.instance.subscribe(resolve as any));
-        widgetManagerPromise
-            .then((wm) => {
+        function reInitializeWidgetManager(resolve?: (value: WidgetManager) => void) {
```

Comment:
We could end up with new widget managers, the subscription should listen to new changes and re-initialize the promise.

Reply:
The function seems weird? At least made it harder for me to parse

Would this be easier to understand?

```
widgetManagerPromise = Promise.resolve((resolve) => WidgetManager.instance.subscribe((wm) => {
   // your current contents here
}
```